website/integrations: add terraform cloud (#17610)
goauthentik/authentik: website/integrations/infrastructure/terraform-cloud/index.md

    - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.
    - **Choose a Provider type**: select **SAML Provider** as the provider type.
    - **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations:
        - Set the **ACS URL** to `https://temporary.domain`.
        - Set the **Issuer** to `https://authentik.company`.
        - Set the **Service Provider Binding** to `Post`.
        - Set the **Audience** to `https://temporary.domain`.
        - Under **Advanced protocol settings**, select an available **Signing Certificate**.
    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

### Copy the metadata URL

1. Log in to authentik as an administrator and open the authentik Admin interface.

> 2. Navigate to **Applications** > **Providers** and click the application created in the previous step.

Reviewer: This hurt my head. Do you mean go to Applications, and click the name of the app, then click **Related** in the left navigation pane? 

<img width="702" height="555" alt="Image" src="https://github.com/user-attachments/assets/a2863bb2-4baf-44b8-ba80-226099513c0c" />

Author: good catch

3. Under **Related objects**, click **Copy download URL** and take note of the URL as it will be used in the next step.

## Terraform Cloud configuration

1. Log in to [Terraform Cloud](https://app.terraform.io) as an administrator.
2. Select your organization from the dropdown menu in the top left, then click **Settings** > **SSO**.
3. Click **Setup SSO**, click **SAML**, then **Next**, and set the **Metadata URL** to the URL copied in the previous step.
4. Click **Save Settings**.
5. Under **HCP Terraform**, take note of the **Entity ID (Audience)** and **Assertion Consumer URL** values. Do not close this window.

## Configure the remaining information in authentik

1. Log in to authentik as an administrator and open the authentik Admin interface.

> 2. Navigate to **Applications** > **Providers** and click the application created in the previous step.

Reviewer: ```suggestion
2. Navigate to **Applications** > **Providers** and provider that you created in the previous step.
```

Reviewer: It's the provider one needs to edit, no? Not the app.

3. Click **Edit**.
4. Under **Protocol settings**, set the value of the **ACS URL** to the **Assertion Consumer URL** value from the previous step. Then, set the value of the **Audience** to the **Entity ID (Audience)** value from the previous step.
5. Click **Update**.

## Enabling Terraform Cloud SSO

> 1. Back where we left off in the Terraform Cloud SSO settings, click **Test**. Successfully authenticating will result in a green checkmark and **Successful** appearing. 2. Then, to enable SSO, click **Enable**.

Author: oh it might work either even if closed and reopen i'd have to check

2. Read the warning message that appears and click **Enable SAML**.

## Configuration verification

> To verify that authentik is correctly integrated with Terraform Cloud, first log out of Terrafom Cloud. Open the [Terraform Cloud login page](https://app.terraform.io/) and click **Sign in with Terraform SSO**. Enter the name of your organization, click **Next** and you'll be redirected to authentik. Once authenticated, you will be signed into Terraform Cloud.

Reviewer: ```suggestion
To verify that authentik is correctly integrated with Terraform Cloud, first log out of Terrafom Cloud. Open the [Terraform Cloud login page](https://app.terraform.io/) and click **Sign in with Terraform SSO**. Enter the name of your organization, click **Next**, and you'll be redirected to authentik. Once authenticated, you will be signed into Terraform Cloud.
```
